File: setup_verbose.py
import logging

def setup_verbose(config):
    logging.basicConfig(level=logging.DEBUG if config.get("verbose") else logging.INFO)
    return logging.getLogger(__name__)

def read_with_verbose(file_path, config):
    logger = setup_verbose(config)
    logger.debug(f"Checking permissions for {file_path}")
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    logger.debug(f"Reading file {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        return f.read()

File: ts-tools/setup_verbose.ts

import * as winston from 'winston';
import * as fs from 'fs';
import { checkFilePermission } from './check_file_permission';

export function setupVerbose(config: any): winston.Logger {
    const level = config.verbose ? 'debug' : 'info';
    const logger = winston.createLogger({
        level: level,
        format: winston.format.json(),
        transports: [
            new winston.transports.Console({
                format: winston.format.simple(),
            }),
        ],
    });
    return logger;
}

export function readWithVerbose(filePath: string, config: any): Promise<string> {
    const logger = setupVerbose(config);
    logger.debug(`Checking permissions for ${filePath}`);
    return new Promise((resolve, reject) => {
        checkFilePermission(filePath, "read", config)
            .then(() => {
                logger.debug(`Reading file ${filePath}`);
                fs.readFile(filePath, 'utf-8', (err, data) => {
                    if (err) {
                        reject(err);
                    } else {
                        resolve(data);
                    }
                });
            })
            .catch(reject);
    });
}
